File: app/src/main/java/com/example/purepawapp/data/seed/AppBootstrap.java
package com.example.purepawapp.data.seed;

import com.example.purepawapp.di.ServiceLocator;

public final class AppBootstrap {

    public static final String CUSTOMER_EMAIL = "customer@purepaw.vn";
    public static final String CUSTOMER_PASSWORD = "Customer123";
    public static final String ADMIN_EMAIL = "admin@purepaw.vn";
    public static final String ADMIN_PASSWORD = "Admin123";

    private AppBootstrap() {
    }

    public static void runIfNeeded() {
        var sessionManager = ServiceLocator.getSessionManager();
        if (sessionManager.isBootstrapDoneOnce()) return;

        try {
            new FirestoreSeeder(ServiceLocator.getFirestore()).seedAll();
        } catch (Exception ignored) {
        }

        createAccountIfMissing("Nguyễn Văn Khách", CUSTOMER_EMAIL, CUSTOMER_PASSWORD, "0901234567", "user");
        createAccountIfMissing("Quản Trị Viên", ADMIN_EMAIL, ADMIN_PASSWORD, "0909999999", "admin");

        sessionManager.markBootstrapDone();
    }

    private static void createAccountIfMissing(String name, String email, String password, String phone, String role) {
        // Failures (e.g. the account already exists from a previous install) are
        // ignored — this is a best-effort debug convenience, not a critical path.
        ServiceLocator.getAuthRepository().signUp(name, email, password, phone, role, new com.example.purepawapp.data.repository.RepoCallback<>() {
            @Override
            public void onSuccess(String result) {
                ServiceLocator.getAuthRepository().logout();
            }

            @Override
            public void onError(Exception error) {
            }
        });
    }
}

File: app/src/main/java/com/example/purepawapp/data/seed/AppBootstrap.kt
package com.example.purepawapp.data.seed

import com.example.purepawapp.di.ServiceLocator

/**
 * One-time debug-only setup: seeds Firestore demo data and creates two ready-to-use
 * accounts (customer + admin) so the app can be exercised without manual data entry.
 * Runs once per install, gated by [com.example.purepawapp.data.session.SessionManager].
 */
object AppBootstrap {

    const val CUSTOMER_EMAIL = "customer@purepaw.vn"
    const val CUSTOMER_PASSWORD = "Customer123"
    const val ADMIN_EMAIL = "admin@purepaw.vn"
    const val ADMIN_PASSWORD = "Admin123"

    suspend fun runIfNeeded() {
        val sessionManager = ServiceLocator.sessionManager
        if (sessionManager.isBootstrapDoneOnce()) return

        runCatching { FirestoreSeeder(ServiceLocator.firestore).seedAll() }

        createAccountIfMissing(
            name = "Nguyễn Văn Khách",
            email = CUSTOMER_EMAIL,
            password = CUSTOMER_PASSWORD,
            phone = "0901234567",
            role = "user"
        )
        createAccountIfMissing(
            name = "Quản Trị Viên",
            email = ADMIN_EMAIL,
            password = ADMIN_PASSWORD,
            phone = "0909999999",
            role = "admin"
        )

        sessionManager.markBootstrapDone()
    }

    private suspend fun createAccountIfMissing(
        name: String,
        email: String,
        password: String,
        phone: String,
        role: String
    ) {
        // Failures (e.g. the account already exists from a previous install) are
        // ignored — this is a best-effort debug convenience, not a critical path.
        ServiceLocator.authRepository.signUp(name, email, password, phone, role)
            .onSuccess { ServiceLocator.authRepository.logout() }
    }
}
